bblfsh/compat.py

```diff
@@ -0,0 +1,228 @@
+            timeout = int(timeout)
+
+        res = self._bblfsh_cli.parse(filename, language, contents,
+                                     mode=mode, timeout=timeout)
+        return CompatParseResponse(res, filename)
+
+    def parse(self, filename: str, language: str = None, contents: str = None,
+              timeout: float = None) -> CompatParseResponse:
+
+        return self._parse(filename, language, contents, timeout,
+                           Mode.Value('ANNOTATED'))
+
+    def native_parse(self, filename: str, language: str = None,
+                     contents: str = None,
+                     timeout: float = None) -> CompatParseResponse:
+
+        return self._parse(filename, language, contents, timeout,
+                           Mode.Value('NATIVE'))
+
+    def supported_languages(self) -> List[str]:
+        return self._bblfsh_cli.supported_languages()
+
+    def version(self) -> VersionResponse:
+        return self._bblfsh_cli.version()
+
+    def close(self) -> None:
+        return self._bblfsh_cli.close()
+
+
+class CompatNodeIterator:
+    def __init__(
+            self,
+            nodeit: NodeIterator,
+            only_nodes: bool = False
+    ) -> None:
+        self._nodeit = nodeit
+        self._only_nodes = only_nodes
+        # Used to forward calls of the old Node object
+        # Check if this, and properties(), are needed
+        self._last_node: Optional[Node] = None
+
+    def __iter__(self) -> 'CompatNodeIterator':
+        return self
+
+    def __next__(self) -> Node:
+        next_val = next(self._nodeit)
+
+        is_node = isinstance(next_val, Node)
+        val = next_val.internal_node if is_node else next_val
+
+        # Skip positions and non dicts/lists, the later if only_nodes = True
+        skip = False
+        if isinstance(val, dict):
+            if "@type" not in val or val["@type"] == "uast:Positions":
+                skip = True
+        elif self._only_nodes:
+            skip = True
+
+        if skip:
+            val = self.__next__().internal_node
+
+        ret_val = next_val if is_node else Node(value=val)
+        self._last_node = ret_val
+        return ret_val
+
+    def filter(self, query: str) -> Optional['CompatNodeIterator']:
+        if not self._last_node:
+            return None
+
+        return filter(self._last_node, query)
+
+    @property
+    def properties(self) -> dict:
+        if isinstance(self._last_node, dict):
+            return self._last_node.keys()
+        else:
+            return {}
+
+
+def iterator(n: Union[Node, CompatNodeIterator], order: TreeOrder = TreeOrder.PRE_ORDER)\
+        -> CompatNodeIterator:
+
+    if isinstance(n, CompatNodeIterator):
+        return CompatNodeIterator(n._nodeit.iterate(order), only_nodes=True)
+    elif isinstance(n, Node):
+        nat_it = native_iterator(n.internal_node, order)
+        return CompatNodeIterator(NodeIterator(nat_it), only_nodes=True)
+    elif isinstance(n, dict):
```

Comment:
Assuming the type annotations were actually checked, would this even be possible?

Reply:
It's possible since the new API can work with normal dictionaries as nodes, but the `Union` parameter annotation was wrong since it was missing the dict; I'll fix it on the next commit.

Reply:
Done!